### PR TITLE
Fix issue with git diff

### PR DESCRIPTION
**Pull request by haroldsphinx**

### CODE REVIEW

File: .github/workflows/simnode.yml [hunk 1]
```diff
@@ -22,12 +22,16 @@ jobs:
...
+
+      - name: Get branch name
+        id: branch-name
+        uses: tj-actions/branch-names@v5.1
 
       - name: Check for runtime changes
         id: check_runtime
         env:
           BASE_BRANCH: ${{ github.event.pull_request.base.ref }}
+          GITHUB_BRANCH_NAME: ${{ steps.branch-name.outputs.current_branch }}
```
Comment (seunlanlege) [inline .github/workflows/simnode.yml:34]: we also need thiis forr `.maintain/run_simnode.sh`

Comment (seunlanlege) [reply .github/workflows/simnode.yml:34]: and benchmarks

File: .maintain/client_release.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ RELEASE_VERSION=$(git tag --sort=committerdate | grep -E '^v[0-9]' | tail -1)
 # last two tags
 PREV_TAG=$(gh release list -L=2 | sed -n '2 p' | awk '{print $(NF-1)}')
 
-if has_client_changes "${PREV_TAG}" "${GITHUB_REF_NAME}"; then
+if has_client_changes "${PREV_TAG}" "${GITHUB_BRANCH_NAME}"; then
```
Comment (seunlanlege) [inline .maintain/client_release.sh:18]: nahh, release pipeline works on push not pull_request

File: .maintain/common/lib.sh [hunk 3]
```diff
@@ -1,6 +1,10 @@
 #!/bin/bash
 
-GITHUB_REF_NAME=$(git rev-parse --abbrev-ref HEAD)
+echo "make sure the main branch and release tag are available in shallow clones"
+git fetch --depth="${GIT_DEPTH:-100}" origin "${BASE_BRANCH}"
+git fetch --depth="${GIT_DEPTH:-100}" origin main
+git fetch --depth="${GIT_DEPTH:-100}" origin releases
+git fetch --depth="${GIT_DEPTH:-100}" origin "${GITHUB_BRANCH_NAME}"
```
Comment (seunlanlege) [inline .maintain/common/lib.sh:7]: doubt we need this

File: .maintain/runtime_release.sh [hunk 4]
```diff
@@ -38,7 +38,7 @@ build_runtime() {
 for i in "${VERSIONS_FILES[@]}"; do
   while IFS=',' read -r output chain folder; do
     echo "check if the wasm sources changed for $chain"
-    if has_runtime_changes "${PREV_TAG}" "${GITHUB_REF_NAME}" "$folder"; then
+    if has_runtime_changes "${PREV_TAG}" "${GITHUB_BRANCH_NAME}" "$folder"; then
```
Comment (seunlanlege) [inline .maintain/runtime_release.sh:41]: this also works on push not pull_request

File: .github/workflows/simnode.yml (hunk 1 above)
Comment (haroldsphinx) [reply .github/workflows/simnode.yml:34]: Updated

File: .maintain/client_release.sh (hunk 2 above)
Comment (haroldsphinx) [reply .maintain/client_release.sh:18]: Updated

File: .maintain/common/lib.sh (hunk 3 above)
Comment (haroldsphinx) [reply .maintain/common/lib.sh:7]: Removed

File: .maintain/runtime_release.sh (hunk 4 above)
Comment (haroldsphinx) [reply .maintain/runtime_release.sh:41]: Updated

**Pull request merged**